### PR TITLE
Adds more tests

### PR DESCRIPTION
**Pull request by roshkhatri**
This PR forks all the redis tests that were running on the [docker-library/official-images](https://github.com/docker-library/official-images).

I have added these as the external tests while we're using the [bashbrew](https://github.com/docker-library/bashbrew) scripts to build, run, test and push it.

### CODE REVIEW

**Comment by roshkhatri** [comment]
@inetol  @jrwren would you be interested in reviewing this PR?

**Comment by roshkhatri** [comment]
@yosifkit @tianon would you please take a look at this PR?

**Comment by roshkhatri** [comment]
@inetol  @jrwren would you be interested in reviewing this PR?

**Review by jrwren — APPROVED** [review]
LGTM

If I were nit picking I'd say add the missing trailing newlines and use [[ over [ consistently.

**Review by inetol — COMMENTED** [review]
Being a *fork* that removed everything except the specific tests for Valkey there are some files that could be deleted like `config.sh`, even more important is a deprecated Docker flag as `--link` that limits my ability to run the tests under Podman as well as some ShellCheck errors that would be convenient to fix.

I have opened a PR to address these issues against this branch [here](https://github.com/roshkhatri/valkey-container/pull/2).

**Comment by tianon** [comment]
I'm not sure why you're copying _all_ the tests, especially without changes; the bashbrew GitHub action will run the tests in `config.sh` in addition to the upstream DOI tests (which you can also reference in your config).

You can also explicitly set `GITHUB_REPOSITORY` to control the image name (https://github.com/tianon/docker-qemu/blob/72cd4ebd938b1628a5e6d0abbd2ceac1590aef88/.github/workflows/ci.yml#L34).

You might also want to switch from `@HEAD` to the latest tagged release of bashbrew so you're not alpha testing our changes like we prefer to. 😄

**Comment by roshkhatri** [comment]
The reason I am doing this is, in future we might be moving away and we might want to modify the tests and add new ones.
I thought that it would have been better to keep the tests and the code in the same repo for maintainability. 

WDYT @tianon, does it makes sense?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     outputs:
       strategy: ${{ steps.generate-jobs.outputs.strategy }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: docker-library/bashbrew@HEAD
```
Comment (roshkhatri) [inline .github/workflows/ci.yml]: will update to latest release.
```suggestion
      - uses: docker-library/bashbrew@v0.1.12
```

**Review by madolson — APPROVED** [review]

**Pull request merged**